### ISSUE TITLE
Missing feature "Anonymize contact"

### ISSUE THREAD

**Issue by ekkeguembel** [mautic/mautic#14119]
### Mautic Series

7.x series

### Mautic installed version

7.0

### Way of installing

I downloaded a release from https://www.mautic.org/mautic-releases

### PHP version

8.1.0

### What happened?

Creating this as reference for upcoming PR.

# Goal(s)

Public core patch for extra GDPR compliance

# Background information

So far, if a user desires deletion (which for some is implicit with opt-out), we can only delete the user entirely. Problem is that this also deletes all statistics etc.

# Optional: Desired outcome

# Verifiable Acceptance Criteria

* a new Campaign action “Anonymize contact” exists
* the related dialog “Anonymize contact”  is similar to “delete contact” (red text: Permanently deletes personal information of the contact. Warning: this is irreversible!)
* but it adds some additional options:
  - “Pseudonymize” toggle (default = off) with Info text: “Pseudonymization will turn the personal data into a one-way hash of itself (using a SHA256 hash) instead of overwriting with random data. This means that if someone has the original personal data, they can run the same hash and thus re-identify this contact.”
  - Fields to delete (Multiselect list of contact field labels. Must not include unique fields.)
  - Fields to anonymize (Multiselect list of contact fields labels. offer only fields of type text or email)
* If user selects the same field in both multiselects → give error message “A field cannot be both deleted and anonymized”, ideally followed by the custom field label
Example: A field cannot be both deleted and anonymized: Email, Last Name
* When the campaign action is run for a contact, delete the selected fields in the leads table (set 0 / NULL / … depending on the field type)
* When the campaign action is run for a contact, anonymize / pseudonymize the selected fields in the leads table (set 256-byte random value id Pseudonymize=0, or set SHA256(original value) if Pseudonymize=1)
* if type=email , add @pseudo.nym or @ano.nym to the 256 bytes string
*  If leads.email was selected, do the same for (prefix_)
   - All form_results of the contact (identified via form_submissions are completely removed  anonymized
   - all audit log entries for the leads are deleted
   - Sentence “All audit log entries for the leads will also be deleted.” in the campaign action UI (below field selectors)
# Out of scope / Not in this version
* covering fields added by other plugins

### How can we reproduce this issue?

:) 

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<br /><hr>
Care about this issue? Want to get it resolved sooner? If you are a <a href='https://www.mautic.org/become-a-member-of-mautic'>member of Mautic</a>, you can add some funds to the <a href='https://opencollective.com/mautic/projects/bounties'>Bounties Project</a> so that the person who completes this task can claim those funds once it is merged by a member of the core team! Read the docs <a href='https://contribute.mautic.org/product-team/mautic-bounty-programme'>here.</a>

**Comment by RCheesley** [comment]
Thanks for the issue and for the PR which resolves it!
